assets/js/atomic/blocks/product-elements/image/block.tsx

```diff
@@ -17,39 +16,74 @@ import {
 } from '@woocommerce/base-hooks';
 import { withProductDataContext } from '@woocommerce/shared-hocs';
 import { useStoreEvents } from '@woocommerce/base-context/hooks';
+import type { HTMLAttributes } from 'react';
 
 /**
  * Internal dependencies
  */
 import ProductSaleBadge from '../sale-badge/block';
 import './style.scss';
+import type { BlockAttributes } from './types';
 
-/**
- * Product Image Block Component.
- *
- * @param {Object}            props                           Incoming props.
- * @param {string}            [props.className]               CSS Class name for the component.
- * @param {string|undefined}  [props.imageSizing]             Size of image to use.
- * @param {boolean|undefined} [props.showProductLink]         Whether or not to display a link to the product page.
- * @param {boolean}           [props.showSaleBadge]           Whether or not to display the on sale badge.
- * @param {string|undefined}  [props.saleBadgeAlign]          How should the sale badge be aligned if displayed.
- * @param {boolean}           [props.isDescendentOfQueryLoop] Whether or not be a children of Query Loop Block.
- * @return {*} The component.
- */
```

Comment:
It would be useful to retain this documentation within the `types.ts`.

Reply:
I thought we would no longer need JSDocs when using TS. Do you have an example for me where we document the purpose of props when using TS? I guess your primary intension is explaining the purpose of the attributes, isn't it?

Reply:
I mean, TS allows you to dispense with all the typing and naming in JSDoc, but you can still use parameters descriptions to help developers. So I would argue that, whenever possible, we should retain. Of course, there are more useful descriptions and less useful descriptions. Here are some examples:

* https://github.dev/woocommerce/woocommerce-blocks/blob/21d7d6672279188338dbde38a153c8c68fcedb7d/assets/js/blocks/product-query/types.ts#L11
* https://github.dev/woocommerce/woocommerce-blocks/blob/21d7d6672279188338dbde38a153c8c68fcedb7d/assets/js/base/components/button/index.tsx#L14
* https://github.dev/woocommerce/woocommerce-blocks/blob/21d7d6672279188338dbde38a153c8c68fcedb7d/assets/js/base/components/country-input/CountryInputProps.ts#L1
* https://github.dev/woocommerce/woocommerce-blocks/blob/21d7d6672279188338dbde38a153c8c68fcedb7d/assets/js/base/components/pagination/index.tsx#L14
* https://github.dev/woocommerce/woocommerce-blocks/blob/21d7d6672279188338dbde38a153c8c68fcedb7d/assets/js/base/components/price-slider/index.tsx#L1

I swear not all of them are by me! 😂

Reply:
Thanks for clarifying this, @sunyatasattva. I documented the types accordingly.